### PR TITLE
feat: Zoom and pan images in the viewer

### PR DESCRIPTION
**Pull request by y-lohse**

### CODE REVIEW

**Comment by y-lohse** [comment]
[(3) [PHO] Ajout du zoom et du drag à la visionneuse sur mobile](https://trello.com/c/QyYO31j2/367-3-pho-ajout-du-zoom-et-du-drag-%C3%A0-la-visionneuse-sur-mobile)

File: src/components/Viewer.jsx [hunk 1]
```diff
@@ -39,11 +57,148 @@ export class Viewer extends Component {
...
+
+    this.gesturesHandler.get('pinch').set({ enable: true })
+    this.gesturesHandler.get('pan').set({ direction: Hammer.DIRECTION_ALL })
+
+    // During a gesture, everything is computed with a base value (the state of the image when the gesture starts) and a delta (a translation / zoom, described by the gesture). When a gesture starts, we record the current state of the image.
+    this.gesturesHandler.on('panstart', this.prepareForGesture.bind(this))
+    this.gesturesHandler.on('pinchstart', this.prepareForGesture.bind(this))
+
+    // during a pan, we add the gestures delta to the initial offset to get the new offset. The new offset is then scaled : if the pan distance was 100px, but the image was scaled 2x, the actual offset should only be 50px. FInally, this value is clamped to make sure the user can't pan further than the edges.
+    this.gesturesHandler.on('pan', e => {
```
Comment (enguerran) [inline src/components/Viewer.jsx]: The more I see comments that try to express the code behavior, the more I think it should be abstract into a library.

**Comment by y-lohse** [comment]
I considered moving the pan and zoom code into a HOC, but ultimately I decided against it because there are a few places where the photo viewer and this code are tightly coupled:

- the toolbar is hidden when a gesture starts
- the swipes are disabled when there's a zoom
- it needs access to the wrapper and the photo DOM elements

It's *possible* to make these things generic and package everything into a library, but I don't think it would make the code clearer. IMO the pan & zoom code is related to the photo viewer, and only the photo viewer, so that's where it should be.

File: src/components/Viewer.jsx [hunk 2]
```diff
@@ -12,6 +12,12 @@ import ImageLoader from './ImageLoader'
 const KEY_CODE_LEFT = 37
 const KEY_CODE_RIGHT = 39
 const TOOLBAR_HIDE_DELAY = 3000
+const MIN_SCALE = 1
+const MAX_SCALE = 6
+const MASS = 13       // Determines how "heavy" the photo feels when paning the image. The bigger the number, the heavier it is.
```
Comment (y-lohse) [inline src/components/Viewer.jsx]: Fair point, I'll try to describe what it affects more clearly.

**Comment by enguerran** [comment]
I like when we apply SRP, and the component here has way too much responsibilities.
But if you did consider every possible solution, I can swallow back my SRP needs.

**Pull request merged**